src/controller/postRouter.js

```diff
@@ -10,15 +10,83 @@ const postRouter = express.Router();
 postRouter.post("/", authenticate, validateTextBody, validateScore, async (req, res) => {
     //TODO check song title exists in API
     try {
-        await createPost(res.locals.user.username, req.body.text, req.body.score, req.body.title);
-        res.status(200).json({
-            message: "Post successfully created"
-        });
+        const post = await createPost(res.locals.user.username, req.body.text, req.body.score, req.body.title);
+        res.status(200).json(post);
     } catch (err) {
         handleServiceError(err, res);
     }
 });
 
+postRouter.patch("/:id", authenticate, async (req, res) => {
+    const {id} = req.params;
+    try {
+        const post = await getPost(id);
+        const {user} = res.locals;
+        let {flag} = req.body;
+        if (flag !== undefined && post.postedBy === user.username) {
+            flag = undefined; // Users cannot flag/unflag their own post
+        }
+        if (flag !== undefined && (typeof(flag) !== "number" || (flag > 1 || flag < 0))) {
+            return res.status(400).json({message: "provided flag must be a number (0 or 1)"});
+        }
+
+        if (user.role === "user" && post.postedBy !== user.username) {
+            //Can't update, can only flag if not an admin or the poster
+            if (flag === undefined) {
+                return res.status(400).json({message: "flag must be provided in body"});
+            }
+            await updatePostFlag(id, flag);
+            return res.status(200).json({id, updated: {isFlagged: flag}})
+        } else {
+            // Only get updatable fields from the body
+            const {description, title, score} = req.body;
+            if (description === undefined && title === undefined && score === undefined && flag === undefined) {
+                return res.status(400).json({message: "No updatable attributes provided. Must provide description, title, flag, or score in body (flag is not valid if you are the poster)"});
+            }
+            if (score && typeof(score) !== "number") {
```

Comment:
I kind of disagree with this because these statements aren't data changing logic. They're just guard clauses that exit immediately if we know the given parameters will cause issues which I tend to check immediately because I was always taught to fail fast when I write code. Nonetheless, I will move the checks to the service layer from now on

Reply:
I can understand where your coming from, I generally like keeping strict boundaries for what an object/function can do because it helps code locality and because I'm used to following the Single Responsibility Principle. But doing an early out is a valid choice as well.